common/src/margins.rs

```diff
@@ -138,50 +138,86 @@ pub fn initialize_margins(dimension: u8, voxels: &mut VoxelData) {
     }
 }
 
-/// Assuming that the voxel at `chunk` and `coords` is set to `material`, updates the cooresponding
-/// margin in the chunk at direction `direction` from `chunk` if such a margin exists. Unpopulated chunks
-/// are ignored.
-pub fn update_margin_voxel(
+/// Based on the given `coords` and the neighboring voxel at direction
+/// `direction` (if it's in a different chunk), updates both of their respective
+/// margins to match each others' materials.
+pub fn reconcile_margin_voxels(
     graph: &mut Graph,
     chunk: ChunkId,
     coords: Coords,
     direction: ChunkDirection,
-    material: Material,
 ) {
-    let coords: CoordsWithMargins = coords.into();
+    let coords_of_edge_voxel: CoordsWithMargins = coords.into();
     let dimension = graph.layout().dimension();
-    let edge_coord = match direction.sign {
-        CoordSign::Plus => dimension,
-        CoordSign::Minus => 1,
-    };
-    if coords[direction.axis] != edge_coord {
-        // There is nothing to do if we're not on an edge voxel.
+
+    // There is nothing to do if we're not on an edge voxel.
```

Comment:
Face, not edge, right?

Reply:
Fair. I think my use of `edge_coord` instead of something like `boundary_coord` was a mistake. I'll fix this up in multiple places.

I'm happy to say "face voxel" or "boundary voxel" here. The former is more specific, while the latter would be more consistent with `boundary_coord`. (I don't want to say `face_coord` because that seems like a less suitable name for a single coordinate value).